### ISSUE TITLE
Possible extension virus and reporting security isssues ?

### ISSUE THREAD

**Issue by xnnkmd** [Microsoft/vscode#33782]
Where do I report possible security issues regarding 3rd party extensions that Microsoft promote in your marketplace ?

Specificly, I have installed vscode with the gitlens extension and get this error:

Scan type: Auto-Protect Scan
Event: Risk Found!
Security risk detected: VBS.Downloader.Trojan
File: C:\Users\xxx\.vscode\extensions\eamodio.gitlens-4.4.2\node_modules\copy-paste\platform\fallbacks\paste.vbs
Location: C:\Users\xxx\.vscode\extensions\eamodio.gitlens-4.4.2\node_modules\copy-paste\platform\fallbacks

**Comment by jrieken** [comment]
Thanks, this is the right place. We will investigate.

**Comment by xnnkmd** [comment]
Thanks for looking at this quickly. It would properly be good if the vscode team published on your page how to deal with urgent secruity issues like this (I was looking but could find nothing about security).

**Comment by xnnkmd** [comment]
BTW: Issue was found by Symantec Endpoint Protection v. 12.1.6

**Comment by cg-atorste** [comment]
Seems like this file is triggering the AV alert:

https://github.com/xavi-/node-copy-paste/blob/master/platform/fallbacks/paste.vbs

As far as I can tell, the script is perfectly benign, and this is just a false positive from Symantec. According to Virustotal, Symantec is the only AV that triggers from this file:

https://virustotal.com/#/file/142020fff889cbaff71795ce8e3e8649576dfed6ca57ead6f208fb58edef782d/detection

This npm module seems to be relatively widely used in VSCode extensions, I got the detections from "donjayamanne.python-0.7.0" and "humao.rest-client-0.15.1".

**Comment by xnnkmd** [comment]
@jrieken I am a bit disappointed that there is no answer yet from MS. I would have thought issues like this had a very high priority ?

**Comment by JeffreyFish** [comment]
Hi,
I met this issue too. Could you please fix this issue in a high priority?

**Comment by gyzmau** [comment]
To add some voice, I got the same issue with the Paste.vbs and Symantec (And sure more and more will get it).
For the record detected into donjayamanne.python-0.7.0 and ms-mssql.mssql-1.1.0

But not sure it's really an issue with vscode itself.
It looks like the Paste.vbs is not new.(https://github.com/xavi-/node-copy-paste/blob/master/platform/fallbacks/paste.vbs)

Symantec update his definition, few hours ago for me, it might coming from them.

**Comment by mishafeingold** [comment]
Yeah I'm getting the same thing. Virus detected with Symantec Endpoint Protection for Mac v12.1.6
 Infection name: VBS.Downloader.Trojan located in: 
~/.vscode/extensions/donjayamanne.python-0.7.0/node_modules/copy-paste/platform/fallbacks/paste.vbs

**Comment by YAMNSheff** [comment]
Same thing here with Symantec Endpoint Protection 12.1.5.

%USERPROFILE%\.vscode\extensions\ms-sql.mssql-1.1.0\node_modules\copy-paste\platform\fallbacks\paste.vbs

**Comment by cg-atorste** [comment]
I don't think "me too" comments are too useful here. I'm not even sure if VScode can do anything here; I'd take this discussion first to your AV vendor, it is their false positive afterall. Has anyone succeeded at reporting this at https://submit.symantec.com/false_positive/? I tried, but I was continuously blocked by the AV so that didn't work

Then if that is not effective then maybe go to the individual extensions or even the upstream node module and try to get it changed so that it doesn't trigger the alert.

**Comment by gyzmau** [comment]
Hi,

I reported the issue to Symantec yesterday.
They replied me: *We have been unable to reproduce the detection issue described in your submission.*

In the meantime i just restored the file from Symantec and rescan it again. And didn't have the issue.

I think it can be closed.

**Comment by BlackyWolf** [comment]
Hi,

We were discussing this yesterday at work and one of our compliance engineers sent the paste.vbs file to be analyzed by Symantec. They confirmed it to be an actual trojan. However, like @gyzmau said, it seems it got fixed at some point?

**Comment by jbeanky** [comment]
@BlackyWolf same here. The current download doesn't appear to be infected. Perhaps it was infected for a short period of time. Maybe the npm account was hacked?

**Comment by xnnkmd** [comment]
Hmm. Still radio silence from the Microsoft VS team. Guess we have to report this to somebody else at Microsoft that takes such matters seriously.

**Comment by Jimmy0987** [comment]
I caught the same situation as @xnnkmd encountered 3days ago.
In my case, file is D:\Application\VSCode-win32-x64-1.16.0-insider\resources\app\extensions\azure-account\node_modules\copy-paste\platform\fallbacks\paste.vbs

Result of analysis by virustotal was the same as the link below : Symantec was the only AV that triggers from this file

https://virustotal.com/#/file/142020fff889cbaff71795ce8e3e8649576dfed6ca57ead6f208fb58edef782d/detection

So the speculation by @BlackyWolf and @jbeanky might be doubtful.  If they are right, Microsoft should take this issue more seriously as @xnnkmd said.  I think this phenomenon might be due to false positive by Symantec as @cg-atorste speculates.

**Comment by seanmcbreen** [comment]
Hi All,

Apologies for the delay here - @xnnkmd I had a response ready yesterday and forgot to post it 👎 

We take issue like this very seriously and I think this has pointed out a few gaps in our process - there should be an easier way to get to the marketplace team for any issue such as this - we have an alias that is for this purpose VSMarketplace@microsoft.com I've added this to our documentation and I'm working with the marketplace to make this more prominent.

A mail to that alias would key off a quick investigation to see if the extension in question included a virus and if so it would be taken out of the marketplace.  The marketplace team are always looking to improve their process and this is actually something they are actively working on.  they use tools much like the one @Jimmy0987 pointed out above.

So we should do a better job of making an escalation route visible.  I'll be adding a FAQ item for this today.

Now at the same time @cg-atorste is correct in that the only real recourse in a situation like this is to report the (in this case) false positive to Symantec.  This needs to come from one of their customers unfortunately (not the VS Code team).  

False positives are unfortunately more common than you would suspect - for instance this has happened several times to VS Code itself.  In all cases these have been false positives and we have had to work with the reporting users to get them to contact their AV Vendor.

Net, net sorry for the delay in coming back - we are listening and items like this are very important so thanks for bringing it to our attention.  I'll ensure we update out FAQ with some improved guidance as well.

Closing this issue for now - but if we need to re-open LMK

Sean
VS Code Team member
